demos/python_demos/face_recognition_demo.py

```diff
@@ -0,0 +1,1037 @@
+        self.active_requests = 0
+
+        self.flush_outputs()
+
+    def deploy(self, device, context, queue_size=1):
+        self.context = context
+        self.max_requests = queue_size
+        self.device_model = context.deploy_model(
+            self.model, device, self.max_requests)
+        self.model = None
+
+    def enqueue(self, input):
+        self.flush_outputs()
+
+        if self.max_requests <= self.active_requests:
+            log.warn("Processing request rejected - too much requests")
+            return False
+
+        self.device_model.start_async(self.active_requests, input)
+        self.active_requests += 1
+        return True
+
+    def wait(self):
+        if self.active_requests <= 0:
+            return
+
+        self.perf_stats = [None, ] * self.active_requests
+        self.outputs = [None, ] * self.active_requests
+        for i in range(self.active_requests):
+            self.device_model.requests[i].wait()
+            self.outputs[i] = self.device_model.requests[i].outputs
+            self.perf_stats[i] = self.device_model.requests[i].get_perf_counts()
+
+        self.active_requests = 0
+
+    def get_outputs(self):
+        self.wait()
+        return self.outputs
+
+    def get_performance_stats(self):
+        return self.perf_stats
+
+    def flush_outputs(self):
+        self.perf_stats = []
+        self.outputs = []
+
+    def _resize(self, frame, target_shape):
+        assert len(frame.shape) == len(target_shape), \
+            "Expected a frame with %s dimensions, but got %s" % \
+            (len(target_shape), len(frame.shape))
+
+        assert frame.shape[0] == 1, "Only batch size 1 is supported"
+        n, c, h, w = target_shape
+
+        input = frame[0]
+        if not np.array_equal(target_shape[-2:], frame.shape[-2:]):
+            input = input.transpose((1, 2, 0)) # to HWC
+            input = cv2.resize(input, (w, h))
+            input = input.transpose((2, 0, 1)) # to CHW
+
+        return input.reshape((n, c, h, w))
+
+
+class FaceDetector(Module):
+    class Result:
+        OUTPUT_SIZE = 7
+
+        def __init__(self, output):
+            self.image_id = output[0]
+            self.label = int(output[1])
+            self.confidence = output[2]
+            self.position = np.array((output[3], output[4])) # x, y
+            self.size = np.array((output[5], output[6])) # width, height
+
+    def __init__(self, model, confidence_threshold=0.5, roi_scale_factor=1.15):
+        super(FaceDetector, self).__init__(model)
+
+        assert len(model.inputs) == 1, "Expected 1 input blob"
+        assert len(model.outputs) == 1, "Expected 1 output blob"
+        self.input_blob = next(iter(model.inputs))
+        self.output_blob = next(iter(model.outputs))
+        self.input_shape = model.inputs[self.input_blob].shape
+        self.output_shape = model.outputs[self.output_blob].shape
+
+        assert np.array_equal([1, 3, 300, 300], self.input_shape), \
```

Comment:
I'm not sure that here we should check input shapes against **hardcoded** values. Generally speaking, this sample can be executed with `intel_models/face-detection-adas-0001` model, which has **384x672** input resolution.

Reply:
I'm not saying that this should hold for other models. I'm saying that this is True for `face-detector` and it's better to add `intel_models/face-detection-adas-0001` as 1 more choice for face detector.